apps/05Mastermind.js

```diff
@@ -29,12 +30,47 @@ function getRandomInt(min, max) {
   return Math.floor(Math.random() * (max - min)) + min;
 }
 
-function generateHint() {
+function generateHint(solution, guess) {
   // your code here
+  var solutionArray = solution.split('');
+  var guessArray = guess.split('');
+  var correctLetterLocations = 0;
+  var correctLetters = 0;
+  for (var i = 0; i < solutionArray.length; i++) {
+    if (solutionArray[i] === guessArray[i]) {
+      correctLetterLocations++;
+      solutionArray[i] = null;  
+    }
+    
+  }
+  for (i = 0; i <= solutionArray.length; i++ ) {
+    var targetIndex = guessArray.indexOf(solutionArray[i]);
+   
+    if (targetIndex > -1) {
+      correctLetters++;
+      solutionArray[i] = null;
+     
+    }
+  }
+  
+
+  return colors.red(correctLetterLocations) + "-" + colors.white(correctLetters);
 }
 
 function mastermind(guess) {
   // your code here
+  solution = 'abcd';
```

Comment:
A solution is already generated when after getPrompt() is called. 

No need to declare one here.